File: Data Structure/Hash/Easy/LET_no1_Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # return empty list? => no
        # notice that the same element cannot be used twice.
        
        # Approach:
        # use hashmap to record (key, value) pair -> key: num, value: index
        # if num is picked up, find out whether num - k is in hashmap
        
        memo = dict()    # Space: O(n)
        for i, num in enumerate(nums):    # Time: O(n)
            if target - num in memo:
                return [memo[target - num], i]
            memo[num] = i
        
        return []

File: Data Structure/Hash/Easy/LET_no1_Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> memo;
        for (int i = 0; i < nums.size(); i++) {
            int aim = target - nums[i];
            if (memo.count(aim))
                return {memo[aim], i};
            memo[nums[i]] = i;
        }

        return {};
    }
};